contracts/InterestRateModels/CheckpointRateModel.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.25;
+
+import { ensureNonzeroAddress } from "@venusprotocol/solidity-utilities/contracts/validators.sol";
+import { InterestRateModel } from "./InterestRateModel.sol";
+
+/**
+ * @title Venus CheckpointRateModel Contract
+ * @notice A contract that applies a rate model based on whether a certain checkpoint
+ *   in time has passed. Useful to adjust interest rates for block time changes in
+ *   the underlying networks, if the block time update is scheduled in advance.
+ * @author Venus
+ */
+contract CheckpointRateModel is InterestRateModel {
```

Comment:
`JumpRateModelV2` inherits `TimeManagerV8` too. Should `CheckpointRateModel` implement the public views, to offer the same interface?

Reply:
I don't think so. It's a different rate model, so it's ok for it to not implement the same interface as `JumpRateModelV2`, as long as it implements `InterestRateModel`. We can have a checkpoint rate model migrating from JumpRateModel to TwoKinksRateModel at a certain timestamp, there's no restriction that the two rate model kinds must be the same.